homeassistant/components/systemmonitor/binary_sensor.py

```diff
@@ -144,6 +138,24 @@ def __init__(
         )
         self.argument = argument
 
+    async def async_added_to_hass(self) -> None:
+        """When added to hass."""
+        if TYPE_CHECKING:
+            assert self.unique_id
+        self.coordinator.update_subscribers[
+            self.entity_description.add_to_update(self)
+        ].add(self.unique_id)
+        return await super().async_added_to_hass()
+
+    async def async_will_remove_from_hass(self) -> None:
+        """When removed from hass."""
+        if TYPE_CHECKING:
+            assert self.unique_id
+        self.coordinator.update_subscribers[
+            self.entity_description.add_to_update(self)
+        ].discard(self.unique_id)
```

Comment:
The entity id would work as a unique value as well since it cannot change while its added to hass, and it saves you the type checking block